src/components/trainings/trainingCard3.component.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import {
+  Image,
+  View,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  ImageProps,
+} from 'react-native';
+import { Exercise } from '@src/core/model/exercise.model';
+import {
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+import { Button } from '@kitten/ui';
+import { Text } from '@src/components/common';
+import {
+  ClockIcon,
+  FlashIcon,
+} from '@src/assets/icons';
+
+interface ComponentProps {
+  training: Exercise;
+  index?: number;
+  onDetails: (index: number) => void;
+  onTiming: (index: number) => void;
+  onEnergy: (index: number) => void;
+}
+
+export type TrainingCardProps = ThemedComponentProps & ComponentProps & TouchableOpacityProps;
+
+class TrainingCard3Component extends React.Component<TrainingCardProps> {
+
+  private onDetails = (): void => {
+    this.props.onDetails(this.props.index);
```

Comment:
prefer calling button press handlers by `on[name]ButtonPress` pattern

Reply:
Needs discussion because this way is not common used in production. Such detailing should be in the "change" methods. In other cases, conciseness is welcome. In any case, we can make a naming guide for the project and discuss each bullet. Not refactored.